src/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -34,16 +37,36 @@
  *
  * Tactics are stateful, and use Primitives to implement their behaviour. They also
  * make heavy use of our Evaluation functions in order to help them make decisions.
+ *
+ * @tparam TacticFsm The Tactic FSM to base this tactic off of (e.g. AttackerTactic needs
+ * AttackerFSM)
+ * @tparam TacticSubFsms the sub FSMs this tactic uses (e.g. AttackerTactic needs
+ * DribbleFSM as a sub FSM)
  */
-class Tactic
+template <class TacticFsm, class... TacticSubFsms>
+class Tactic : public TacticInterface
```

Comment:
I'd suggest renaming `Tactic` to `BaseTactic` (along with my other comment about renaming `TacticInterface` to `Tactic`)